packages/pulseaudio/module-sles-sink.c

```diff
@@ -239,99 +290,81 @@ static void process_rewind(pa_sink *s) {
-    u->sles_msgq = pa_asyncmsgq_new(0);
-    if (!u->sles_msgq) {
+    if (!(u->sles_msgq = pa_asyncmsgq_new(0))) {
         pa_log("pa_asyncmsgq_new() failed.");
         goto fail;
     }
-
-    /* The msgq from the AudioTrack RT thread should have an even higher
-     * priority than the normal message queues, to match the guarantee
-     * all other drivers make: supplying the audio device with data is
-     * the top priority -- and as long as that is possible we don't do
-     * anything else */
     u->rtpoll_item = pa_rtpoll_item_new_asyncmsgq_read(u->rtpoll, PA_RTPOLL_EARLY-1, u->sles_msgq);
 
-    if (!(ma = pa_modargs_new(m->argument, valid_modargs))) {
-        pa_log("Failed to parse module arguments.");
-        goto fail;
-    }
+    pa_sink_new_data_init(&sink_data);
+    sink_data.driver = __FILE__;
+    sink_data.module = m;
 
     ss = m->core->default_sample_spec;
-    pa_channel_map_init_stereo(&map);
-
-    switch (ss.format) {
-    case PA_SAMPLE_S16LE:
-    case PA_SAMPLE_S24LE:
-    case PA_SAMPLE_S32LE:
-    case PA_SAMPLE_FLOAT32LE:
-        break;
-    default:
-        pa_log("Sample format not supported");
+    if (pa_modargs_get_sample_rate(ma, &ss.rate) < 0) {
+        pa_log("Invalid rate specification.");
         goto fail;
     }
-    pa_modargs_get_sample_rate(ma, &ss.rate);
+
+    pa_channel_map_init_stereo(&map);
     ss.channels = map.channels;
 
-    if (pa_init_sles_player(u, &ss) < 0)
+    if (pa_init_sles_player(u, &ss))
         goto fail;
 
-    pa_sink_new_data_init(&data);
-    data.driver = __FILE__;
-    data.module = m;
-    pa_sink_new_data_set_name(&data, pa_modargs_get_value(ma, "sink_name", DEFAULT_SINK_NAME));
-    pa_sink_new_data_set_sample_spec(&data, &ss);
-    pa_sink_new_data_set_channel_map(&data, &map);
-    pa_proplist_sets(data.proplist, PA_PROP_DEVICE_DESCRIPTION, _("OpenSL ES Output"));
-    pa_proplist_sets(data.proplist, PA_PROP_DEVICE_CLASS, "abstract");
-
-    if (pa_modargs_get_proplist(ma, "sink_properties", data.proplist, PA_UPDATE_REPLACE) < 0) {
-        pa_log("Invalid properties");
-        pa_sink_new_data_done(&data);
+    pa_sink_new_data_set_name(&sink_data, pa_modargs_get_value(ma, "sink_name", DEFAULT_SINK_NAME));
+    pa_sink_new_data_set_sample_spec(&sink_data, &ss);
+    pa_sink_new_data_set_channel_map(&sink_data, &map);
+    pa_proplist_sets(sink_data.proplist, PA_PROP_DEVICE_DESCRIPTION, "OpenSL ES Output");
+    pa_proplist_sets(sink_data.proplist, PA_PROP_DEVICE_CLASS, "abstract");
+
+    if (pa_modargs_get_proplist(ma, "sink_properties", sink_data.proplist, PA_UPDATE_REPLACE) < 0) {
+        pa_log("Invalid properties.");
+        pa_sink_new_data_done(&sink_data);
         goto fail;
     }
 
-    u->sink = pa_sink_new(m->core, &data, 0);
-    pa_sink_new_data_done(&data);
-
-    if (!u->sink) {
+    if (!(u->sink = pa_sink_new(m->core, &sink_data, 0))) {
         pa_log("Failed to create sink object.");
+        pa_sink_new_data_done(&sink_data);
```

Comment:
Why repeat this to allow the assignment in condition style? Isn't it a bad style anyway?

Reply:
This matches what PA itself does in most of its module code (I don't like it either, tbh), but I can move it out if you want.